### PR TITLE
CLI Baseline Initial Commit

### PR DESCRIPTION
**Pull request by Tomdye**

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = function (grunt) {
+	require('grunt-dojo2').initConfig(grunt, {
+		intern: {
+			options: {
+				nodeOptions: [
+					'--harmony',
```
Comment (agubler) [inline Gruntfile.js]: Do we still need the `--harmony` flags?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  "name": "dojo-cli",
+  "version": "0.0.3-pre",
+  "description": "Dojo 2 CLI utility",
+  "homepage": "http://dojotoolkit.org",
+  "bugs": {
+    "url": "https://github.com/dojo/cli/issues"
+  },
+  "license": "BSD-3-Clause",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (agubler) [inline package.json]: we have some tests right?

File: Gruntfile.js (hunk 1 above)
Comment (Tomdye) [inline Gruntfile.js]: with Node 6.3 I still needed these two I believe. It complained that the `destructuring` one no longer existed but was happy about these.

File: package.json (hunk 2 above)
Comment (Tomdye) [inline package.json]: writing them now

File: package.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    "postinstall": "typings install"
```
Comment (agubler) [inline package.json]: Why is this `postInstall` and not in the build pipeline as per other modules?

Comment (Tomdye) [inline package.json]: this was before the other changes, if you point me in the right direction i will change

File: src/commands/newApp.ts [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+	const renderResponses = await Promise.all(renderPromises);
+	const renderDisplayCount = chalk.green(renderResponses.length.toString());
+
+	log('info', chalk.bold.green('Summary: ') + `Rendered: ${renderDisplayCount}`);
+};
+
+function getSelectedModuleConfig(selectedModuleIds: string[], availableModuleConfig: ModuleConfigMap): ModuleConfigMap {
+	let modules: ModuleConfigMap = {};
+
+	// Get just the module config we care about
```
Comment (agubler) [inline src/commands/newApp.ts]: do we need to have a comment here?

Comment (Tomdye) [inline src/commands/newApp.ts]: no

File: Gruntfile.js (hunk 1 above)
Comment (novemberborn) [inline Gruntfile.js]: [Default parameters should be supported](http://node.green/#default-function-parameters).

Is there anything specific that `--harmony` enables? Would be good to enable just those.

**Pull request closed without merging**